src/lib.rs

```diff
@@ -38,27 +47,85 @@ impl<const N: usize> Default for MuxTimer<N> {
 
 impl<const N: usize> MuxTimer<N> {
     /// Fire timer for event with `ordinal` after `timeout` duration.
-    /// Returns `true` if the timer was armed, `false` if it was already armed for the same event with sooner deadline.
-    pub fn fire_after(self: Pin<&mut Self>, ordinal: impl Into<usize>, timeout: Duration) -> bool {
-        self.fire_at(ordinal, Instant::now() + timeout)
+    /// Returns `true` if the timer was armed, `false` if it was already armed for the same event and provided `CoalesceMode`.
+    pub fn fire_after(
+        self: Pin<&mut Self>,
+        ordinal: impl Into<usize>,
+        timeout: Duration,
+        coalesce_mode: CoalesceMode,
+    ) -> bool {
+        self.fire_at(ordinal, Instant::now() + timeout, coalesce_mode)
     }
 
     /// Fire timer for event with `ordinal` at `deadline`.
-    /// Returns `true` if the timer was armed, `false` if it was already armed for the same event with sooner deadline.
-    pub fn fire_at(self: Pin<&mut Self>, ordinal: impl Into<usize>, deadline: Instant) -> bool {
+    /// Returns `true` if the timer was armed, `false` if it was already armed for the same event and provided `CoalesceMode`.
+    pub fn fire_at(
+        self: Pin<&mut Self>,
+        ordinal: impl Into<usize>,
+        deadline: Instant,
+        coalesce_mode: CoalesceMode,
+    ) -> bool {
         let ordinal = ordinal.into();
-        if self.deadlines[ordinal].is_some_and(|d| d < deadline) {
+        if self.deadlines[ordinal].is_some_and(|d| match coalesce_mode {
+            CoalesceMode::Earliest => d < deadline,
+            CoalesceMode::Latest => d > deadline,
+        }) {
             return false;
         }
+
         let current_deadline = self.deadline();
         let mut this = self.project();
         this.deadlines[ordinal] = Some(deadline);
-        if current_deadline.map_or(true, |d| deadline < d) {
-            this.arm(ordinal, deadline);
+
+        match coalesce_mode {
+            CoalesceMode::Earliest => {
+                if current_deadline.map_or(true, |d| deadline < d) {
```

Comment:
Rust 1.82 finally added `is_none_or` which would work here, but would mean a pretty high MSRV so we can keep this for now!